ciao-controller/internal/datastore/datastore.go

```diff
@@ -1220,6 +1220,17 @@ func (ds *Datastore) AddNode(nodeID string, nodeType payloads.Resource) error {
 	return nil
 }
 
+// GetNode retrieves a node in the node cache.
+func (ds *Datastore) GetNode(nodeID string) (types.Node, error) {
+	var node types.Node
+
```

Comment:
Could be.  The lock is stuck next to the map in the Datastore struct.  Doesn't say it's required though.  Searching through the code it does look like it's used to protect the map.